ddtrace/settings/config.py

```diff
@@ -187,6 +196,72 @@ def get_error_ranges(error_range_str):
     return error_ranges  # type: ignore[return-value]
 
 
+ConfigSource = Literal["default", "env", "code"]
+
+
+class _ConfigItem:
+    """Configuration item that tracks the value of a setting, and where it came from."""
+
+    def __init__(self, name, default, envs):
+        self._name = name
+        self._default_value = default
+        self._env_value = None
+        self._code_value = None
+        self._envs = envs
+        for (env_var, parser) in envs:
+            if env_var in os.environ:
+                self._env_value = parser(os.environ[env_var])
```

Comment:
Should we introduce an `_env_name` field to track which env_var was set?

Reply:
great idea! I like the idea of tracking which variable ended up being used.

I will track this as follow-up work in addition to adding documentation strings, usage examples, etc.